src/sage/combinat/permutation.py

```diff
@@ -5383,6 +5383,41 @@ def shifted_shuffle(self, other):
         return self.shifted_concatenation(other, "right").\
         right_permutohedron_interval(self.shifted_concatenation(other, "left"))
 
+    def transposition_distance(self, other):
+        r"""
+        Compute the transposition distance between two permutations.
+
+        INPUT:
+
+        - ``other`` -- Another permutation to compute the transposition distance to.
+
+        OUTPUT:
+
+        - An integer representing the transposition distance between the two permutations.
+
+        EXAMPLES::
+
+            sage: p1 = Permutation([2, 1, 3, 4])
+            sage: p2 = Permutation([1, 3, 2, 4])
+            sage: p1.transposition_distance(p2)
+            2
+        """
+        other = self.parent()(other)
```

Comment:
Sorry, what I said was not quite correct. This should find a common parent for both. There is the `@coerce_binop` decorator that will do this automatically. Then this line won't be necessary and it will work for both ways. You should have doctests covering these cases as well,.